modules/client/src/interfaces.ts

```diff
@@ -259,20 +262,34 @@ export enum TransferType {
 export enum TokenType {
   NATIVE = "native",
   ERC20 = "erc20",
+  ERC721 = "erc721",
 }
 
 type BaseTokenTransfer = {
-  amount: bigint;
   creationDate: Date;
   transactionId: string;
+  to: string;
+  from: string;
 };
 
 type NativeTokenTransfer = BaseTokenTransfer & {
   tokenType: TokenType.NATIVE;
+  amount: bigint;
+  reference: string;
+};
+
+type Erc721TokenTransfer = BaseTokenTransfer & {
+  tokenType: TokenType.ERC721;
+  token: {
+    address: string;
+    name: string;
+    symbol: string;
+  };
 };
 
 type Erc20TokenTransfer = BaseTokenTransfer & {
   tokenType: TokenType.ERC20;
+  amount: bigint;
```

Comment:
ERC721 transfers have more fields than represented here. At the very least there is the ID of the token that you are transferring the ownership to someone else

I would personally leave a TO DO here, because this is not the focus now

Reply:
Yes forgot to add the address, my bad

Reply:
What I mean is that when you transfer an NFT... (i.e. the crypto punk number 23) you are transferring a specific thing, not a simple amount.

You are transferring the ownership, which means that `amount` is not going to work in general

Reply:
This is ERC20 typing only, ERC721 does not have amount